src/api/exchange/ExchangeApi.ts

```diff
@@ -202,40 +183,14 @@ export class ExchangeApiImpl extends DepositApiImpl implements ExchangeApi {
     logDebug(`[ExchangeApiImpl] subscribing to trade events for address ${userAddress} and networkId ${networkId}`)
 
     const subscription = await this.getTradeSubscription(params)
-    // Two ways of subscribing
 
-    // 1: with error handling
-    subscription.subscribe((error, event) => {
-      if (error) {
+    subscription
+      .on('data', event => callback(this.parseTradeEvent(event)))
+      .on('error', error => {
```

Comment:
So I see you prefer 2 independent subscriptions instead of 1 for both.
Any performance reason or just because you like it?

Reply:
Previously `on('error'` wouldn't trigger for creation subscription error
Now no difference